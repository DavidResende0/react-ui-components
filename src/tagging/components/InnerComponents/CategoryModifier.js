import React from 'react';
import PropTypes from 'prop-types';
import { Col, ControlLabel, FormGroup } from 'patternfly-react';
import { __ } from '../../../global-functions';
import TagSelector from './TagSelector';
import TaggingPropTypes from '../TaggingPropTypes';

const CategoryModifier = ({
  tagCategories,
  onTagCategoryChange,
  selectedTagCategory,
  categoryLabel,
}) => (
  <FormGroup>
    <Col xs={12} md={4} lg={4}>
      <ControlLabel>{categoryLabel}</ControlLabel>
    </Col>
    <Col xs={12} md={8} lg={8}>
      <TagSelector
        tagCategories={tagCategories}
        onTagCategoryChange={onTagCategoryChange}
        selectedOption={selectedTagCategory}
      />
    </Col>
  </FormGroup>
);

CategoryModifier.propTypes = {
  tagCategories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired).isRequired,
  selectedTagCategory: TaggingPropTypes.category,
  onTagCategoryChange: PropTypes.func.isRequired,
  categoryLabel: PropTypes.string,
};

CategoryModifier.defaultProps = {
  categoryLabel: __('Category'),
  selectedTagCategory: {},
};

export default CategoryModifier;
