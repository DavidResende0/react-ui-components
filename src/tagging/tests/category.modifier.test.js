import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CategoryModifier from '../components/categoryModifier';

const selectedTagCategory = { description: 'animal', id: 1 };
const onTagCategoryChange = jest.fn();
const tagCategories = [
  { description: 'Name', id: 1 },
  { description: 'Number', id: 2 },
  { description: 'Animal', id: 3 },
  { description: 'Food', id: 4 },
];

describe('TagCategory Component', () => {
  it('match snapshot', () => {
    const tree = mount(<CategoryModifier
      selectedTagCategory={selectedTagCategory}
      onTagCategoryChange={onTagCategoryChange}
      tagCategories={tagCategories}
    />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
