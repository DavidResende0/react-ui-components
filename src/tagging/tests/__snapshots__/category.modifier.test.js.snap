// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCategory Component match snapshot 1`] = `
<CategoryModifier
  categoryLabel="Category"
  onTagCategoryChange={[MockFunction]}
  selectedTagCategory={
    Object {
      "description": "animal",
      "id": 1,
    }
  }
  tagCategories={
    Array [
      Object {
        "description": "Name",
        "id": 1,
      },
      Object {
        "description": "Number",
        "id": 2,
      },
      Object {
        "description": "Animal",
        "id": 3,
      },
      Object {
        "description": "Food",
        "id": 4,
      },
    ]
  }
>
  <FormGroup
    bsClass="form-group"
  >
    <div
      className="form-group"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={4}
        xs={12}
      >
        <div
          className="col-lg-6 col-md-4 col-xs-12"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
            >
              Category
            </label>
          </ControlLabel>
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={8}
        xs={12}
      >
        <div
          className="col-lg-6 col-md-8 col-xs-12"
        >
          <TagSelector
            infoText="Only a single value can be assigned from these categories"
            onTagCategoryChange={[MockFunction]}
            selectedOption={
              Object {
                "description": "animal",
                "id": 1,
              }
            }
            tagCategories={
              Array [
                Object {
                  "description": "Name",
                  "id": 1,
                },
                Object {
                  "description": "Number",
                  "id": 2,
                },
                Object {
                  "description": "Animal",
                  "id": 3,
                },
                Object {
                  "description": "Food",
                  "id": 4,
                },
              ]
            }
          >
            <Select
              arrowRenderer={[Function]}
              autosize={true}
              backspaceRemoves={true}
              backspaceToRemoveMessage="Press backspace to remove {label}"
              className="selected-option"
              clearAllText="Clear all"
              clearRenderer={[Function]}
              clearValueText="Clear value"
              clearable={false}
              closeOnSelect={true}
              deleteRemoves={true}
              delimiter=","
              disabled={false}
              escapeClearsValue={true}
              filterOptions={[Function]}
              ignoreAccents={true}
              ignoreCase={true}
              inputProps={Object {}}
              isLoading={false}
              joinValues={false}
              label="animal"
              labelKey="label"
              matchPos="any"
              matchProp="any"
              menuBuffer={0}
              menuRenderer={[Function]}
              multi={false}
              name="form-field-name"
              noResultsText="No results found"
              onBlurResetsInput={true}
              onChange={[Function]}
              onCloseResetsInput={true}
              onSelectResetsInput={true}
              openOnClick={true}
              optionComponent={[Function]}
              options={
                Array [
                  Object {
                    "label": "Name",
                    "value": 1,
                  },
                  Object {
                    "label": "Number",
                    "value": 2,
                  },
                  Object {
                    "label": "Animal",
                    "value": 3,
                  },
                  Object {
                    "label": "Food",
                    "value": 4,
                  },
                ]
              }
              pageSize={5}
              placeholder="Select..."
              removeSelected={true}
              required={false}
              rtl={false}
              scrollMenuIntoView={true}
              searchable={true}
              simpleValue={false}
              tabSelectsValue={true}
              trimFilter={true}
              value={1}
              valueComponent={[Function]}
              valueKey="value"
            >
              <div
                className="Select selected-option has-value is-searchable Select--single"
              >
                <input
                  disabled={false}
                  key="hidden.0"
                  name="form-field-name"
                  type="hidden"
                  value="1"
                />
                <div
                  className="Select-control"
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                >
                  <span
                    className="Select-multi-value-wrapper"
                    id="react-select-2--value"
                  >
                    <Value
                      disabled={false}
                      id="react-select-2--value-item"
                      instancePrefix="react-select-2-"
                      onClick={null}
                      placeholder="Select..."
                      value={
                        Object {
                          "label": "Name",
                          "value": 1,
                        }
                      }
                    >
                      <div
                        className="Select-value"
                      >
                        <span
                          aria-selected="true"
                          className="Select-value-label"
                          id="react-select-2--value-item"
                          role="option"
                        >
                          Name
                        </span>
                      </div>
                    </Value>
                    <AutosizeInput
                      aria-activedescendant="react-select-2--value"
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-owns=""
                      className="Select-input"
                      injectStyles={true}
                      minWidth="5"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      value=""
                    >
                      <div
                        className="Select-input"
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-activedescendant="react-select-2--value"
                          aria-expanded="false"
                          aria-haspopup="false"
                          aria-owns=""
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          role="combobox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "5px",
                            }
                          }
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        />
                      </div>
                    </AutosizeInput>
                  </span>
                  <span
                    className="Select-arrow-zone"
                    onMouseDown={[Function]}
                  >
                    <span
                      className="Select-arrow"
                      onMouseDown={[Function]}
                    />
                  </span>
                </div>
              </div>
            </Select>
          </TagSelector>
        </div>
      </Col>
    </div>
  </FormGroup>
</CategoryModifier>
`;
